lib/charms/vault_k8s/v0/vault_client.py

```diff
@@ -175,6 +167,34 @@ def set_pki_intermediate_ca_certificate(self, certificate: str, mount: str) -> N
         )
         logger.info("Set the intermediate CA certificate for the PKI backend")
 
+    def sign_pki_certificate_signing_request(
+        self, mount: str, role: str, csr: str, common_name: str
+    ) -> Tuple[str, str, List[str]]:
```

Comment:
Because this is in a lib, will this be used elsewhere? I think we might want to consider using a `dataclass` or `namedtuple` as the return type here, to make the API a bit cleaner, similar to what was done in TLS V3.

Reply:
It will be used here and it the vault machine charm.